### PR TITLE
update ingest date index name processor with runnable CONSOLE examples

### PR DESCRIPTION
**Pull request by talevy**
added some more doc examples for the date index name processor.

Relates to #18160

### CODE REVIEW

File: docs/reference/ingest/ingest-node.asciidoc [hunk 1]
```diff
@@ -782,32 +782,121 @@ date in the `date1` field:
...
+      {
+        "date_index_name" : {
+          "field" : "date1",
+          "index_name_prefix" : "myindex-",
+          "date_rounding" : "M"
+        }
+      },
+      {
+        "set" : {
+          "field" : "_ingest.timestamp",
```
Comment (talevy) [inline docs/reference/ingest/ingest-node.asciidoc]: This extraneous processor to make the CONSOLE test response consistent made me question the existence of the `_ingest.timestamp` field in general. I am not sure how valuable its use-cases are. If someone wants to understand how long things take, we have pipeline stats now.

what do you think @javanna?

Comment (javanna) [inline docs/reference/ingest/ingest-node.asciidoc]: I don't think that the timestamp should be used to measure how long things take, although that is what people tend to be doing. This was introduced as the timestamp field got deprecated and removed, it allows to add to a document a timestamp that tells when it entered the system, that's it.

Comment (talevy) [inline docs/reference/ingest/ingest-node.asciidoc]: fair enough. I'll leave it as is.

mind reviewing the rest of the PR @javanna?

Comment (nik9000) [inline docs/reference/ingest/ingest-node.asciidoc]: Is the problem that the timestamp would be dependent on the current time without the processor? If so you can actually work around it with `// TESTRESPONSE[s/2016-04-25T12:02:01.789Z/$body.docs.1.doc._ingest.timestamp]` or something like that. The `s///` syntax doctors the text in the docs into the replacement when it generates the actual test. The `$body.docs.1....` syntax is a stash lookup from the response, replacing what we match against with text from the response.

Comment (talevy) [inline docs/reference/ingest/ingest-node.asciidoc]: oooo fancy! that is what I want. I'll update with this. thanks

**Comment by talevy** [comment]
updated to use `s///` syntax in test response for the simulate example

**Comment by nik9000** [comment]
Awesome. LGTM.

**Comment by nik9000** [comment]
@talevy I edited the description to link to #18160. I think this change caught two errors too! Nice!

**Pull request merged**